service/matching/matcher_test.go

```diff
@@ -228,9 +228,11 @@ func (t *MatcherTestSuite) TestRejectSyncMatchWhenBacklog() {
 	go func() {
 		defer wg.Done()
 		for {
+			t.rootMatcher.backlogTasksLock.RLock()
 			if len(t.rootMatcher.backlogTasksCreateTime) > 0 {
 				return // Exit if a task has been added
 			}
+			t.rootMatcher.backlogTasksLock.RUnlock()
 		}
 	}()
 	wg.Wait()
```

Comment:
Now that we changed to lock, why not just an Eventually on rootMatcher.isBacklogNegligible instead of accessing the internals of matcher?

Reply:
Actually i'm not sure if we'd need a rwlock with the eventually approach.